### PR TITLE
perf(losses): stop pinning target logits through the fused ce backward

### PR DESCRIPTION
**Pull request by WindChimeRan**
## Summary

Fused CE retained the full target-logit tensor through backward even though backward only needs the target argmax cached during forward. This PR saves `None` for CE and guards target-pointer use behind the non-CE specialization.

At the measured 32k-vocabulary configuration (`max_anchors=512`, `block_size=16`, bf16), this releases 0.49 GiB. Without a vocabulary mapping, training uses the full verifier vocabulary; for Qwen3's 151,936-token vocabulary, the same configuration releases 2.32 GiB.

The saving scales with `max_anchors × block_size × vocabulary size × element size`, not packed `total_seq_len` at a fixed anchor count. Losses and gradients are unchanged. Distribution losses are unchanged, and compound losses containing one will still retain targets.

Follow-up to #951.

## Results

GB10, 512 anchors × 16 positions × 32k mapped vocabulary:

| | Before | After |
| --- | ---: | ---: |
| Held into backward | 1.31 GiB | 0.83 GiB |
| Backward-phase peak | 1.80 GiB | 1.31 GiB |

Gradient checksum was bit-identical; KL was unchanged.

## Tests

- `pytest tests/unit/models/test_fused_losses.py` — 16 passed
- `pytest tests/unit/models tests/unit/train` — 475 passed, 5 skipped

## Checklist

- [x] The purpose of the PR
- [x] The test plan/results
- [ ] (Optional) The necessary documentation update
- [x] I (a human) have written or reviewed the code in this PR to the best of my ability

### CODE REVIEW

File: tests/unit/models/test_fused_losses.py
Comment (rahul-tuli) [inline tests/unit/models/test_fused_losses.py:1]: The new test checks that backward runs, but could it also spot-check that `logits.grad` matches the eager CE gradient? Without that, the test proves the kernel does not crash with `None` but not that it produces correct results when targets are released early. Even a loose `assert_close` against the eager path would strengthen the coverage.

File: src/speculators/losses/fused.py
Comment (rahul-tuli) [inline src/speculators/losses/fused.py:249]: Minor: the PR description mentions savings of 0.49-2.32 GiB; is that the backward-phase reduction only, or the full forward+backward peak? Since `targets_flat` is still materialized during forward, the peak during forward is unchanged?

File: src/speculators/losses/fused.py [hunk 1]
```diff
@@ -189,7 +189,6 @@ def loss_backward_kernel(
     """Recompute probabilities from the saved stats and apply the OP gradient."""
     pid = tl.program_id(0).to(tl.int64)
     logits_ptr += pid * n_cols
-    targets_ptr += pid * n_cols
```
Comment (rahul-tuli) [inline src/speculators/losses/fused.py:192]: 💯 nice, Just to confirm my reading: in the original code, this line was already dead for CE (no downstream load), so the change is purely structural, right?

**Review by rahul-tuli — COMMENTED** [review]
LGTM, pending comments!

File: src/speculators/losses/fused.py (hunk 1 above)
Comment (WindChimeRan) [reply src/speculators/losses/fused.py:192]: yes!

File: src/speculators/losses/fused.py
Comment (WindChimeRan) [reply src/speculators/losses/fused.py:249]: Yes. it's backward-phase reduction only.

File: tests/unit/models/test_fused_losses.py
Comment (WindChimeRan) [reply tests/unit/models/test_fused_losses.py:1]: Correctness is already covered by test_fused_matches_eager[ce] from the previous PR. This PR does not add new branches, so it's still covered.

**Comment by WindChimeRan** [comment]
Addressed the comments @rahul-tuli !

**Review by fynnsu — APPROVED** [review]
Thanks!
